### PR TITLE
Fix Solr Auth Binding

### PR DESCRIPTION
**Pull request by nickumia-reisys**
Related to https://github.com/GSA/datagov-deploy/issues/3523

### CODE REVIEW

File: manifest.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ service_definitions:
 parameters: []
 required_env_variables: []
 env_config_mapping:
-  SOLR_SERVER: solr.server
+  SOLR_CP_SERVER: solr.server
```
Comment (mogul) [inline manifest.yml:33]: The env var name shouldn't change... The broker only ever needs the control plane endpoint, and it'll be confusing for people configuring the brokerpak if the name of the env var and the name of the config file entry are different! (Plus this change will [break existing staging/prod configuration which uses the old name](https://github.com/GSA/datagov-ssb/blob/7261e274181c02a67f33187bca22d11e5f31ac25/app-setup-solr.sh#L25).)

**Review by mogul — CHANGES_REQUESTED** [review]
I'm still unclear on what the fix actually was... Curling from inside the Solr container? Changing the default Solr image in use? Please explain!

File: manifest.yml (hunk 1 above)
Comment (nickumia-reisys) [reply manifest.yml]: You are right.  I think I changed it not understanding the rework you did to separate KiND from EKS stuff.  I changed it back 👍

**Review by FuhuXia — APPROVED** [review]

File: examples.json-template [hunk 2]
```diff
@@ -5,7 +5,8 @@
     "service_name": "solr-cloud",
     "service_id": "b9013a91-9ce8-4c18-8035-a135a8cd6ff9",
     "plan_id": "e35e9675-413f-4f42-83de-ad5003357e77",
-    "provision_params": {"solrJavaMem":"-Xms300m -Xmx300m", "solrMem":"1G", "solrCpu":"1000m", "cloud_name":"demo"},
+    "provision_params": {"solrJavaMem":"-Xms300m -Xmx300m", "solrMem":"1G", "solrCpu":"1000m", "cloud_name":"demo", "solrImageRepo":"ghcr.io/gsa/catalog.data.gov.solr", "solrImageTag":"8-curl"},
```
Comment (nickumia-reisys) [inline examples.json-template:8]: Use the new image with curl for tests.

File: generate-env.sh [hunk 3]
```diff
@@ -3,33 +3,48 @@
...
+    SOLR_TOKEN=$(kubectl get secret $(kubectl get secrets | grep -oh "default-token-[a-z]*\s") -o json | jq .data.token | tr -d '"')
+    if [[ "$SOLR_TOKEN" == "null" ]]; then
+        # The format of the secret is different if there are more than one token associated with a secret.
+        # The first token works reliably
+        SOLR_TOKEN=$(kubectl get secret $(kubectl get secrets | grep -oh "default-token-[a-z]*\s") -o json | jq .items[0].data.token | tr -d '"')
+    fi
+else
+    # Otherwise it's the same as the normal server control plane URL
+    SOLR_CP_SERVER=${SOLR_SERVER}
+fi
```
Comment (nickumia-reisys) [inline generate-env.sh:28]: Set up repo to cooperate with either a KiND cluster or real EKS cluster.

File: terraform/bind/main.tf [hunk 4]
```diff
@@ -70,23 +70,20 @@ resource "null_resource" "manage_solr_user" {
     # Can't reuse containers because they are left in an unpredictable state after a single run
     # Wait for the command to run before deleting the container
     command = <<-EOF
-      kubectl --kubeconfig <(echo $KUBECONFIG | base64 -d) run temp1 --image=curlimages/curl --  \
+      kubectl --kubeconfig <(echo $KUBECONFIG | base64 -d) exec ${self.triggers.cloud_name}-solrcloud-0 --  curl \
```
Comment (nickumia-reisys) [inline terraform/bind/main.tf:73]: `curl` to Solr API is now performed from within the solrcloud pod itself

File: terraform/provision/main.tf [hunk 5]
```diff
@@ -45,6 +45,7 @@ resource "helm_release" "solrcloud" {
       "replicas"                                   = var.replicas      # How many replicas you want
       "solrOptions.javaMemory"                     = var.solrJavaMem   # How much memory to give each replica
       "solrOptions.security.authenticationType"    = "Basic"
+      "ingressOptions.annotations.nginx\\.ingress\\.kubernetes\\.io/proxy-body-size" = "999m"
```
Comment (nickumia-reisys) [inline terraform/provision/main.tf:48]: Port update from earlier release, https://github.com/GSA/datagov-brokerpak-solr/commit/574336a1cafc30c2940f3678939544d59ebfa139#diff-ba41f8c4953fba843cf840ae370b5681502661992bf51cede0ea49d92187a6aeR45

**Review by nickumia-reisys — COMMENTED** [review]
@mogul I don't know what caused the 'binding' process to fail prior to this PR.  There were a lot of changes to this repo in this PR that may have effected it.  Kubernetes token issues.  Curl unable to reach API.  EKS vs. KiND peculiarities.  Either way, the repo has now been tested with KiND and with a real EKS cluster, so I have confidence that it will work this time.

TODO:  Document how to manually test repo with a real EKS cluster.  The steps are scattered throughout this repo and https://github.com/GSA/datagov-deploy/issues/3523.  It just needs to be consolidated.

Related Repo Update: `curl` was installed in our base solr image https://github.com/GSA/catalog.data.gov/blob/feature/solr8/solr/Dockerfile#L11-L12

**Pull request merged**